prober/main.go

```diff
@@ -0,0 +1,225 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/storage"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pmap/apis/v1alpha1"
+	"github.com/abcxyz/pmap/internal/testhelper"
+	"github.com/abcxyz/pmap/pkg/server"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+const (
+	proberGithubCommitValue    = "prober-test-github-commit"
+	proberGithubRepoValue      = "prober-test-github-repo"
+	proberWorkflowValue        = "prober-test-workflow"
+	proberWorkflowShaValue     = "prober-test-workflow-sha"
+	proberWorkflowRunID        = "prober-test-workflow-id"
+	proberWorkflowRunAttempt   = "1"
+	proberGCSNamePrefix        = "//storage.googleapis.com"
+	proberMappingTraceIDPrefix = "prober-mapping"
+	proberResourceProvider     = "gcp"
+	proberResourceContact      = "pmap-prober@abcxyz.dev"
+)
+
+var (
+	cfg       *config
+	bqClient  *bigquery.Client
+	gcsClient *storage.Client
+)
+
+func main() {
+	ctx, done := signal.NotifyContext(context.Background(),
+		syscall.SIGINT, syscall.SIGTERM)
+	defer done()
+
+	if err := realMain(ctx); err != nil {
+		done()
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+}
+
+func realMain(ctx context.Context) error {
+	logging := logging.FromContext(ctx)
+	// create a global config
+	c, err := newTestConfig(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to parse integration test config: %w", err)
+	}
+	cfg = c
+
+	// create a global gcs client
+	sc, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create gcs client: %w", err)
+	}
+	gcsClient = sc
+	defer sc.Close()
+
+	// create a global bigquery client
+	bc, err := bigquery.NewClient(ctx, c.ProjectID)
+	if err != nil {
+		return fmt.Errorf("failed to create bigquery client: %w", err)
+	}
+	bqClient = bc
+	defer bc.Close()
+
+	ts := strconv.FormatInt(time.Now().Unix(), 10)
+
+	var probeErr error
+	if err := probeMapping(ctx, ts); err != nil {
+		probeErr = errors.Join(probeErr, fmt.Errorf("prober failed for mapping service: %w", err))
+	}
+
+	// TODO: add probePolicy() to probe policy service
+	// errors.Join() is used here so we can join the errors from probePolicy and return them together.
+	// if err := probePolicy(ctx, ts); err != nil {
+	// 	probeErr = errors.Join(probeErr, fmt.Errorf("prober failed for policy service: %w", err))
+	// } else {
+	// 	logging.Info("Policy probe failed.")
+	// }
+	if probeErr == nil {
+		logging.Info("Mapping probe successed")
+	}
```

Comment:
I see now you want to output "mapping success" vs. "policy success". Feel free to use the if/else

Reply:
My current thought only print "Mapping and Policy probe succeed" after probePolicy is added.